### ISSUE TITLE
[feature] update multiple lockfiles if they share modified/built package and package_id

### ISSUE THREAD

**Issue by solvingj** [conan-io/conan#7180]
Summary:

Give users a way to update multiple lockfiles when they share a common package/packageID which has been `"modified/built"`.  Possible ways to approach it include:
* make `conan create` accept multiple lockfiles (probably not the right approach)
* make `conan install` able to detect that the package in the local cache has changed and just update without rebuild (probably not the right approach)
* add `conan graph copy-modified` to copy a status update from one lockfile to N lockfiles, when each of those lockfiles features the same identical packageID for the "Modified/built" node..  (probably the most straightforward). 

Background

During experimentation with CI workflows and lockfiles, there's a major scalability issue with an extremely common scenario.  The premise of the scenario is that many organizations have long lists of products with complicated options, but for which they share binaries for large numbers of low level components built with a common set of conan settings and options.

Example Scenario: 
zlib
 -> openssl
    ->  50 products

* For simplicity, lets imagine our CI just builds a single profile/variant for each product
* Each product has slightly different package options
* All the options used for zlib and openssl are identical between all the profiles
* Thus we have 50 distinct lockfiles

Despite having 50 distinct lockfiles, they all resolve to the same package IDs for zlib and OpenSSL.  If we make a patch to zlib and use revisions and lockfiles to rebuild everything, the current guidance for Conan + CI is: "build zlib once for every product lockfile".  That's 50 builds of zlib and openssl, each of them all having the same package ID as the one before.  This is an unacceptable approach.  Whether run in parallel or sequentially, it's wildly inefficient.  This was just a modest example, if you imagine building 4 or 16 lockfiles for each product, it's many many hundreds of redundant builds of zlib when 4 or 16 would suffice. 

The reason the guidance says "build once with each unique lockfile"  is that the only way to mark a lockfile with "modified/built" under zlib is to run  `conan create --lockfile`.  We need each lockfile to get this update, because they'll all be used when we pass them to their appropriate downstreams for the 50 unique product builds. 

In my local environment, i've used the following trick to workaround this redundant rebuilds by reading the lockfiles and grouping them by the packageID they contain for `zlib`.  Then, I build zlib only once for each unique packageID of zlib found by doing: 

`conan create --lockfile <random_lockfile_from_group>` 

Then, I update the rest of the lockfiles with this trick:

`conan create --lockfile <random_lockfile_from_group> --keep-build --test-folder=None` 

This re-runs the conan create, but skips the actual build and test package.  So, it basically just updates the remaining lockfiles.  It actually does re-run the `package()` method which is ok-ish for now.  IF we had a flag called `--keep-package` , that would be even faster, but, obviously, that's all just workaround optimization. 

The simplest and most desirable solution to this specific problem (given the current implementation of lockfiles), is the one suggested at the start.  Give users a way to update many lockfiles with a single build when all lockfiles share the same package id for the "modified/built" package. 

- [X] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by solvingj** [comment]
I've thought a lot about different possibilities for how this could be implemented.  It's not trivial.  There are a lot of choices to be made.  The required logical steps are: 

1. de-duplicate: given a group of lockfiles return a list of unique packageID's for `current_lib`
2. for-each unique packageID, run conan create one time
3. after each unique packageID has been created, update all related lockfiles with new `prev`

**Scope**

*  Should Conan add features for dealing with groups of lockfiles? 

**Background**
This process of working with groups of lockfiles has more to do with engineering CI outside of Conan than actual package management.  On the other hand, lockfiles are complex domain-specific data structures , and Conan will continue to evolve their implementation details over time.  It's asking a lot of engineering/CI teams to write complex logic for parsing/evaluating/managing them. 

**Smallest Scope**
If Conan chooses not to deal with groups of lockfiles, the simplest way it could help the users for the case above would be something like this: 
`conan lock update --reference zlib/1.0@ --lockfile somefile.lock` 

This could look in local cache for zlib, and update the lockfile with the current revision.  The user would still carry the burden of de-duplicating the lockfiles, choosing an arbitrary one for running the build , and iterating over the rest to run this command.  But, at least it makes the flow possible. 

**Larger Scope**
Given a path to a folder containing lockfiles, or an explicit list of lockfile paths at the command line, do it all for the user.  Conan can identify all the lockfiles which feature the package/package_id in question and update only those ones.  In their CI job, the user will still have to gather all the lockfiles for each of their apps, and propagate the modified lockfiles to downstream jobs as necessary.  However, we've handled the most complex part of the lockfile manipulation within Conan.

**Comment by solvingj** [comment]
Ok, in relation to another issue, @memsharded just informed me that `conan create . --build=outdated` will skip the build of the current package if the "recipe revision" and "package revision" both exist in cache.  So, in theory, this would just be a strictly version of the step to update the additional lockfiles (better than `--keep-build` because it would also bypass the package step.)  

To be clear, this doesn't affect the fundamental "hard part" detailed here: identifying redundant lockfiles and/or updating multiple lockfiles after a single build is still prohibitively complex.

**Comment by memsharded** [comment]
This was implemented in the ``lock bundle`` command in https://github.com/conan-io/conan/pull/8344, will be available in 1.34
